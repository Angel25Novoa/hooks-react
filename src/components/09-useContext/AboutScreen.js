import React, {useContext} from 'react'
import { UserContext } from './UserContext'

const AboutScreen = () => {
  const {user, setUser} = useContext(UserContext) 
  return (
    <div>
      <h1>About Screen</h1>
      <hr />
      <button className="btn btn-info"
      onClick={() => setUser({})}>
        Logout
      </button>
      <pre>{JSON.stringify(user)}</pre>
    </div>
  )
}

export default AboutScreen