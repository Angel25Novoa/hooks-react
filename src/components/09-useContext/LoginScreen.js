import React, {useContext} from 'react'
import { UserContext } from './UserContext'

const LoginScreen = () => {
  //obtener context
  const {user, setUser} = useContext(UserContext) 
  //obtener setUser
  return (
    <div>
      <h1>Login Screen</h1>
      <hr />
      <button className="btn btn-info"
      onClick={() => setUser({
        name: 'Juan',
        email:'',
        id: 2
      })}>
        Login
      </button>
      <pre>{JSON.stringify(user)}</pre>
    </div>
  )
}

export default LoginScreen