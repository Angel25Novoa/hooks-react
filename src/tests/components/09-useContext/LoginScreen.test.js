import React from 'react'
import {mount} from 'enzyme'
import LoginScreen from '../../../components/09-useContext/LoginScreen'
import { UserContext } from '../../../components/09-useContext/UserContext'

describe('Pruebas en el componente LoginScreen', () => {
  const setUser = jest.fn()
  const wrapper = mount( 
    <UserContext.Provider value={{setUser}}>
      <LoginScreen/>
    </UserContext.Provider>
  )
  test('debe de mostrar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('debe ejecutar el setUser con el argumento esperado', () => {
    wrapper.find('button').simulate('click', {
      preventDefault: () => {},
      target: {
        name: 'login',
        value: 'login'
      }
    })
    expect(setUser).toHaveBeenCalledWith({
      name: 'Juan',
      email: '',
      id: 2
    })
  })
})