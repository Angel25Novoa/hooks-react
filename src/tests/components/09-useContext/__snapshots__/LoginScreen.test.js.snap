// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente LoginScreen debe de mostrar el componente correctamente 1`] = `
<div>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-info"
    onClick={[Function]}
  >
    Login
  </button>
  <pre />
</div>
`;
